src/os/bluestore/bluestore_types.h

```diff
@@ -378,7 +378,7 @@ struct bluestore_blob_use_tracker_t {
       bytes_per_au[0] = old_total;
     } else {
       assert(_au_size == au_size);
-      new_len = ROUND_UP_TO(new_len, au_size);
+      new_len = round_up_to<uint32_t>(new_len, au_size);
```

Comment:
please note, the `T` can be deduced from the parameter type. no need to specify it explicitly.

Reply:
what's the disadvantage to specify T? Just curiosity.

Reply:
if there is no disadvantage, it would be better to specify type to prevent:
`
/git/ceph/src/include/intarith.h:80:3: note:   template argument deduction/substitution failed:
/git/ceph/src/os/bluestore/BlueFS.cc:1119:50: note:   deduced conflicting types for parameter ‘T’ (‘long unsigned int’ and ‘unsigned int’)
     length = round_up_to(length, super.block_size);
`

Reply:
i was using this case as an example to raise your attention, not to ask you to apply this to very change in this PR. 

> what's the disadvantage to specify T? Just curiosity.

for instance, you don't need to repeat the type. for some reading fun, see http://en.cppreference.com/w/cpp/language/template_argument_deduction .

Reply:
[OT], could you reply me regarding to https://github.com/ceph/ceph/pull/19222 ? i don't want the efforts to be wasted in this way.